### ISSUE TITLE
Links to tutorial zip archives are incorrect

### ISSUE THREAD

**Issue by jlstevens** [ioam/holoviews#154]
We should find a more robust way of linking to the latest set of notebooks. Maybe just a standard name like ``tutorials-latest.zip`` would do.

I've filed this issue to make sure we have this fixed for the next release!

**Comment by jbednar** [comment]
Having a fixed filename would make sure that the links don't break, but it wouldn't make sure that the .zip file is actually up to date and works with that version.  It might be better to force the new-release-upload process to fail if the correspondingly named zip file for that version does not exist.  That would require making a new copy of the .zip file for every release, but (a) doing that ensures that at least someone has thought about whether any changes are needed for this new release, and (b) shouldn't be costly in git, since I don't think git will actually store two copies of the same identical file even if the names differ.

**Comment by jlstevens** [comment]
> It might be better to force the new-release-upload process to fail if the correspondingly named zip file for that version does not exist.

That would make sense but I don't see it working in practice. The release process uses ``setup.py`` which includes all the current notebooks in the PyPI release (actually, that is another way to get the correct notebooks!) but this is separate from the zip file which goes to a different repository entirely (the ``ioam-github.com`` repository).

Here is an approach that I think is better... maybe we can make the current version into some variable that we can refer to through the docs (must be possible somehow!). Then that version variable is used in the archive name and finally, we make sure there are no broken links on the website using a link checker [like this one](https://validator.w3.org/checklink).

Using this tool I just found one broken link on the website ``http://ipython.org/notebook/`` - there should be no trailing slash.

**Comment by jbednar** [comment]
Ah, I forgot that the .zip is stored outside the repository.  Yes, doing it via a link checker does sound better in this case, and checking for broken links is something we should do before releasing anyway, so that seems like a good approach.  I'm surprised there was only one broken one!

**Comment by jlstevens** [comment]
Annoyingly RST only seems to substitute if there is a space involved:

```rst
Works |VERSION|
Broken_|VERSION|

.. |VERSION| replace:: 1.2.5
```

I can only suggest making a unique definition for the zip archive:

```rst
|ZIPARCHIVE|

.. |ZIPARCHIVE| replace:: tutorials_1.2.5.zip
```

It seems likely that if we remember to update the ``VERSION`` then we won't forget to update ``ZIPARCHIVE`` as they will be defined in the same place.

**Comment by jbednar** [comment]
According to http://docutils.sourceforge.net/docs/ref/rst/directives.html#directives-for-substitution-definitions , you can tell RST to remove the preceding whitespace in a substitution, so that even though the space is required it won't show up after the substitution.

**Comment by jlstevens** [comment]
This seems to work, but then we have to be careful how we use version elsewhere (don't want spaces removed!). Sounds we will still need two possible substitutions.

```rst
archive |VERSION|.zip

.. |VERSION| unicode::  1.2.5
   :ltrim:
```

**Comment by jbednar** [comment]
The trim specification is about the definition of the substitution, not the usage?  That's annoying and not very useful!

**Comment by jlstevens** [comment]
Right. I agree it isn't that helpful but still probably better than encoding a specific file name as a substitution...

**Comment by jbednar** [comment]
Definitely.

**Comment by jlstevens** [comment]
For now, I've uploaded the zip file for 1.2.0 ioam/ioam.github.com@55b50b7 and updated the link dad3f883

**Comment by jlstevens** [comment]
I tried to use the substitutions as suggested but restructured text refused to allow substitutions in links despite all my efforts.

Although that failed, In the end, I implemented an approach that I think will be far more robust:

* No need to change anything in ``conf.py`` unless a new link that depends on the version is added somewhere. The version is now obtained from the declaration of ``setup.py`` - this means the version in the docs will update automatically on release!

* We just need to be careful not to introduce version strings to the docs and remember to define any such links appropriately in ``conf.py``.

This is implemented in 7e83a8d and I think I can close this issue now.
